drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -128,7 +128,7 @@ def process_custom_field(field, value)
     # If this custom field only allows 1 value and there already is one, update it.
     if !cded.repeats && existing_values.exists?
       cde_attributes = { id: existing_values.first.id }
-      if value.present?
+      if value.present? || value.instance_of?(FalseClass)
```

Comment:
can you just?
```suggestion
      unless value.nil?
```

Reply:
I don't really want to save `""` strings, thats the main reason why I left the presence check

Reply:
Is there a better railsy way to do that?

Reply:
Gig I think what you have is fine although maybe it could use a comment.

However I'm a little unsure as to what value might be at this point. Sounds like it could be a string or a boolean? The code suggests that it might also be an array sometimes?

If the value is not well normalized already, maybe we should do that? Please ignore this if it's already the case. Thinking about something like the following case statement tree that will error if we see something unexpected (untested)

```ruby
  def process_field 
    #...
    value = normalize_input_value(cded, value)
    #...
    if value.nil?
      cde_attributes[:_destroy] = 1
    else
      cde_attributes[value_field_name] = value
      cde_attributes.merge!(attrs)
    end
  end

  def normalize_input_value(cded, value)
    ## do we get arrays here?
    return value.map { |v| v.normalize_input_value(cded, v)}.compact.presence if value.is_a?(Array)

    case cded.field_type
    when 'string', 'text'
      case value
      when String
        value&.presence&.strip
      else
        raise "unexpected value \"#{value}\" for #{field}"
      end
    # when 'json', 'integer', 'float', 'date'
    when 'boolean'
      case value
      when nil, ""
        nil
      when true, false
        value
      ## do we get strings here? ("true", "false")
      # when String
      #   ActiveModel::Type::Boolean.new.cast(value)
      else
        raise "unexpected value \"#{value}\" for #{field}"
      end
    else
      raise "unknown field_type #{field_type}"
    end
  end
```

Reply:
@gigxz sorry for the delay, I'm with @ttoomey, this is fine as is, but checking `instance_of?` always rubs me the wrong way, unless there's some specific set of types we expect, and then, I'm with @ttoomey we should handle the different types as a case statement, or at least intentionally.   Happy to just keep it as is, but maybe a comment about why it's checking for a boolean or blank?

Reply:
Thanks, I can normalize it like that, its clearer to read. There is a validation to make sure the definition type and the value type match, so it does error if they dont match.

Reply:
Is `when String` not a code smell but `instance_of?` is? Just wanna get my railsing right here.. to me these seem the same

Reply:
I think feedback is addressed @ttoomey can you take a look?

Reply:
For me it was more of separating the logic from the type determination.  So normalizing the thing you are checking is one operation, then determining validity of the normalized data becomes more of a business operation. It's a very weak smell if you ask me, but definitely something that caught my eye.